compiler/src/iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.cpp

```diff
@@ -528,6 +528,185 @@ wrapOpInDispatchRegion(RewriterBase &rewriter, Operation *op) {
+    SmallVector<Value> &dims = newResultsDynamicDims.emplace_back();
+    if (failed(reifyDynamicResultDims(rewriter, operand.get(), dims))) {
+      return op->emitOpError(
+          "failed to reify dynamic dims of result to be yielded from "
+          "dispatch region");
+    }
+  }
+
+  // Get the new set of dynamic dims without the `op` result dims.
+  SetVector<Value> dynamicDims;
+  for (int64_t res = 0; res < dispatchRegionOp->getNumResults(); ++res) {
+    auto dims = dispatchRegionOp.getResultDynamicDims(res);
+    dynamicDims.insert(dims.begin(), dims.end());
+  }
+  // Create the new dispatch region op. The region does not yet yield the
+  // operands of the `op`.
+  auto newDispatchRegionOp = rewriter.create<IREE::Flow::DispatchRegionOp>(
+      dispatchRegionOp->getLoc(), newResultTypes, dynamicDims.takeVector(),
+      dispatchRegionOp.getWorkload());
+  rewriter.inlineRegionBefore(dispatchRegionOp.getBody(),
+                              newDispatchRegionOp.getBody(),
+                              newDispatchRegionOp.getBody().begin());
+  auto newDispatchReturnOp = cast<IREE::Flow::ReturnOp>(
+      newDispatchRegionOp.getBody().front().getTerminator());
+  {
+    OpBuilder::InsertionGuard returnGuard(rewriter);
+    rewriter.setInsertionPoint(newDispatchReturnOp);
+    rewriter.replaceOpWithNewOp<IREE::Flow::ReturnOp>(newDispatchReturnOp,
+                                                      newReturnedValues);
+  }
+
+  // Append newly yielded operands to the new dispatch results.
+  SmallVector<Value> newYieldedResults = llvm::map_to_vector(
+      yieldedOperands, [](OpOperand *operand) { return operand->get(); });
+  FailureOr<IREE::Flow::DispatchRegionOp> finalDispatchRegionOp =
+      appendDispatchRegionResults(rewriter, newDispatchRegionOp,
+                                  newYieldedResults, newResultsDynamicDims);
+  if (failed(finalDispatchRegionOp)) {
+    return failure();
+  }
+
+  // Replace operands of the `hoistedOp` with dispatch region results.
+  auto replaceHoistedOpOperand = [&](Value v, int operandIdx) {
+    rewriter.modifyOpInPlace(hoistedOp,
+                             [&]() { hoistedOp->setOperand(operandIdx, v); });
+  };
+  for (auto [resIdx, operand] : llvm::enumerate(yieldedOperands)) {
+    auto dispatchResIdx = resIdx + newReturnedValues.size();
+    Value dispatchResult = finalDispatchRegionOp->getResults()[dispatchResIdx];
+    replaceHoistedOpOperand(dispatchResult, operand->getOperandNumber());
+  }
+
+  // Replace the uses of the original dispatch region results with the final
+  // dispatch region results.
+  for (auto [idx, operand] : llvm::enumerate(newDispatchReturnOperands)) {
+    Value result = dispatchRegionOp.getResults()[operand->getOperandNumber()];
+    Value newResult = finalDispatchRegionOp->getResults()[idx];
+    rewriter.replaceAllUsesWith(result, newResult);
+  }
+
+  return hoistedOp;
+}
+
+bool isBroadcastingOp(Operation *op) {
```

Comment:
Do we use `linalg::LinalgOp` type directly? I think we only expect linalg ops and they all implements the interface.